### PR TITLE
Modernise namespaces to use nested specifier

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
Uses modern C++-17 namespaces across the project. The pros of which are:
- Simpler to read
- Less boilerplate
- Style is the same everywhere so less mental burden keeping track

This allows us to use one unified style, so I've updated the developer standards

**To test:**
- Check everything builds
- Explicitly check `ViewPort.cpp` as there was a merge conflict here

*This does not require release notes* because it is not a user visible change

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/core/src/Converters/CloneToNDArray.cpp [hunk 1]
```diff
@@ -19,7 +19,20 @@
 #define NO_IMPORT_ARRAY
 #include <numpy/arrayobject.h>
 
-namespace Mantid::PythonInterface::Converters::Impl {
+namespace Mantid::PythonInterface::Converters {
+#ifdef __APPLE__
```
Comment (martyngigg) [inline Framework/PythonInterface/core/src/Converters/CloneToNDArray.cpp:23]: I'm not sure why this is required on clang. The symbols should be exported correctly...

**Review by martyngigg — DISMISSED** [review]
I'm fully behind these changes. I think the nesting makes it much clearer to see what the full namespace is at a glance

**Comment by gemmaguest** [comment]
@DavidFair there are conflicts

**Comment by martyngigg** [comment]
We've still got conflicts here even after the last resolution unfortuantely.

**Review by martyngigg — DISMISSED** [review]

**Review by rosswhitfield — APPROVED** [review]

**Pull request merged**